# Experimental patch for fixing problem with live locks in thread system
# Fixed random hangs on page load
# need to check on crashes
--- mozilla/widget/src/xpwidgets/nsBaseAppShell.cpp.orig	2007-03-22 08:36:10.000000000 +0200
+++ mozilla/widget/src/xpwidgets/nsBaseAppShell.cpp	2007-03-22 22:06:32.908070250 +0200
@@ -139,21 +139,21 @@ nsBaseAppShell::DoProcessNextNativeEvent
   // If the next native event is not our NativeEventCallback, then we may end
   // up recursing into this function.
   //
   // However, if the next native event is not our NativeEventCallback, but it
   // results in another native event loop, then our NativeEventCallback could
   // fire and it will see mProcessNextNativeEvent as true.
   //
   PRBool prevVal = mProcessingNextNativeEvent;
 
   mProcessingNextNativeEvent = PR_TRUE;
-  PRBool result = ProcessNextNativeEvent(mayWait); 
+  PRBool result = ProcessNextNativeEvent(0); 
   mProcessingNextNativeEvent = prevVal;
   return result;
 }
 
 //-------------------------------------------------------------------------
 // nsIAppShell methods:
 
 NS_IMETHODIMP
 nsBaseAppShell::Run(void)
 {
